pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -248,20 +251,31 @@ func (k *K8sCiliumEndpointsWatcher) endpointUpdated(oldEndpoint, endpoint *types
 			}
 		}
 	}
+
+	// Upsert the labels for the CiliumEndpoint.
+	for _, ip := range ipsAdded {
+		endpointResourceID := ipcachetypes.NewResourceID(ipcachetypes.ResourceKindEndpoint, endpoint.Namespace, endpoint.Name)
+		lbls := labels.NewLabelsFromModel(endpoint.Identity.Labels)
+		ipPrefix := ippkg.IPToNetPrefix(net.ParseIP(ip))
+		k.ipcache.UpsertLabels(ipPrefix, lbls, labels.LabelSourceK8s, endpointResourceID)
```

Comment:
Risk to investigate: I wonder whether this could end up having a significant impact on Cilium's operations for the usual Cilium use case by inadvertently triggering additional local security identity allocation for all workloads in the cluster.

Reply:
Should I investigate adding a separate cache for labels?

Reply:
I think that this information is already in the ipcache, so I'm not quite sure I understand what the missing piece is for your use case.

Reply:
I think the labels are cached only if the data plane is installed, not when the cache is used as standalone. I was wondering if we should add a way to add the labels using the watcher.

Reply:
That doesn't make sense to me, the data plane just registers an `IPIdentityMappingListener` and receives the updates.

Reply:
@joestringer  Thanks for the reply. To clarify, do you mean the `Upsert` call on L235 is already caching the labels, or, do you mean we should pass the labels in that `Upsert` call (on L235) instead of adding it later.

In my testing, I couldn't print the labels without exclusively calling another `Upsert` with labels, but I can find the endpoint name and namespace. I was wondering if we can cache the labels as well so that I can find all required metadata about an IP.